src/terraform-command.js

```diff
@@ -406,16 +479,22 @@ class TerraformCommand extends AbstractCommand {
     Object.keys(config).forEach(hash => {
       const node = config[hash];
 
-      issues[hash] = node.dependsOn.filter(dep => {
-        const key = ConfigLoader.buildComponentHash(dep);
+      issues[hash] = node.dependsOn.filter(name => {
+        const dependentComponent = Object.keys(config).find(it => config[it].name === name);
+        if (!dependentComponent) {
+          return true;
+        }
+
+        const dependentConfig = config[dependentComponent];
+        const key = this._createHashFromName(dependentConfig);
 
         return !config.hasOwnProperty(key);
       });
     });
 
     const messages = Object.keys(issues).filter(it => issues[it].length).map(it => {
-      return `'${config[it].name}' component depends on the component in '${issues[it].join(`', '`)}' ` +
-        `director${issues[it].length > 1 ? 'ies' : 'y'} that doesn't exist`;
+      return `'${config[it].name}' component depends on the component${issues[it].length > 1 ? 's' : ''} ` +
+        `'${issues[it].join(`', '`)}' that doesn't exist`;
```

Comment:
please rename `that doesn't exist` to `that either are not included or do not exist`

Reply:
updated